plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts

```diff
@@ -432,4 +458,56 @@ export class DatabaseNotificationsStore implements NotificationsStore {
   async markUnsaved(options: NotificationModifyOptions): Promise<void> {
     await this.markReadSaved(options.ids, options.user, undefined, null);
   }
+
+  async getUserNotificationOrigins(options: {
+    user: string;
+  }): Promise<string[]> {
+    const rows: { origin: string }[] = await this.db('notification')
```

Comment:
Think it might be worthwhile using `this.db<RowType>` here to make these types easier to use.